pkg/storage/image.go

```diff
@@ -102,18 +104,16 @@ func (svc *imageService) ListImages(systemContext *types.SystemContext, filter s
 			return nil, err
 		}
 		for _, image := range images {
-			var size *uint64
-			if len(image.Names) != 0 {
-				ref, err := svc.getRef(image.Names[0])
-				if err != nil {
-					return nil, err
-				}
-				img, err := ref.NewImage(systemContext)
-				if err != nil {
-					return nil, err
-				}
-				size = imageSize(img)
+			ref, err := istorage.Transport.ParseStoreReference(svc.store, "@"+image.ID)
+			if err != nil {
+				return err
+			}
+			img, err := ref.NewImage(systemContext)
+			if err != nil {
+				return nil, err
 			}
+			size := imageSize(img)
+			img.Close()
```

Comment:
Is the image opened in imageSize or in NewImage?

Reply:
in `NewImage`

Reply:
I can see `img.Close()` in `ImageStatus` as well

Reply:
The `Image` reference is obtained in `NewImage()`.  The `imageSize` function just calls the image's `Size()` method and does some type massaging.